src/pytkdocs/parsers/docstrings/numpy.py

```diff
@@ -0,0 +1,260 @@
+        """
+        parameters = []
+
+        docstring_params = [p for p in docstring_obj.params if p.args[0] == "param"]
+
+        for param in docstring_params:
+            name = param.arg_name
+            kind = None
+            type_name = param.type_name
+            default = param.default or empty
+            try:
+                signature_param = self.context["signature"].parameters[name.lstrip("*")]  # type: ignore
+            except (AttributeError, KeyError):
+                self.error(f"No type annotation for parameter '{name}'")
+            else:
+                if signature_param.annotation is not empty:
+                    type_name = signature_param.annotation
+                if signature_param.default is not empty:
+                    default = signature_param.default
+                kind = signature_param.kind
+            parameters.append(
+                Parameter(
+                    name=param.arg_name,
+                    annotation=type_name,
+                    description=param.description,
+                    default=default,
+                    kind=kind,
+                )
+            )
+
+        if parameters:
+            return Section(Section.Type.PARAMETERS, parameters)
+
+        return None
+
+    def read_attributes_section(
+        self,
+        docstring_obj: docstring_parser.common.Docstring,
+    ) -> Optional[Section]:
+        """
+        Parse an "attributes" section.
+
+        Arguments:
+            docstring_obj: Docstring object parsed by docstring_parser.
+
+        Returns:
+            A tuple containing a `Section` (or `None`).
+        """
+        attributes = []
+        docstring_attributes = [p for p in docstring_obj.params if p.args[0] == "attribute"]
+
+        for attr in docstring_attributes:
+            attributes.append(
+                Attribute(
+                    name=attr.arg_name,
+                    annotation=attr.type_name,
+                    description=attr.description,
+                )
+            )
+
+        if attributes:
+            return Section(Section.Type.ATTRIBUTES, attributes)
+
+        return None
+
+    def read_exceptions_section(
+        self,
+        docstring_obj: docstring_parser.common.Docstring,
+    ) -> Optional[Section]:
+        """
+        Parse an "exceptions" section.
+
+        Arguments:
+            docstring_obj: Docstring object parsed by docstring_parser.
+
+        Returns:
+            A tuple containing a `Section` (or `None`) and the index at which to continue parsing.
+        """
+        exceptions = []
+        except_obj = docstring_obj.raises
+
+        for exception in except_obj:
+            exceptions.append(AnnotatedObject(exception.type_name, exception.description))
+
+        if exceptions:
+            return Section(Section.Type.EXCEPTIONS, exceptions)
+
+        # self.error("Empty exceptions section")
```

Comment:
So, should the error be logged or not? I would say yes, and therefore this should be logged for every empty section. What do you think?

Reply:
Yes I think we could though the docstring_parser library doesn't raise an error or warnings whenever a docstring is malformed, it just returns an empty section. I am now logging an error for empty sections by checking whether the section header is present but no further content can be found.